buffer/buffer.mbt

```diff
@@ -173,6 +173,20 @@ pub fn to_bytes(self : T) -> Bytes {
   Bytes::from_fixedarray(self.data, len=self.len)
 }
 
+///|
+pub fn op_set(self : T, index : Int, value : Byte) -> Unit {
+  let len = self.length()
+  guard index >= 0 && index < len
+  self.data[index] = value
+}
```

Comment:
buffer is an opaque type, when is this API needed?

Reply:
I made a mistake, `buffer` doesn't need random access, just use `bytes`. I think the opaque nature can be retained and I'll close this pr.